azavea/iow-boundary-tool
# Add panes for layers

## Overview

Putting each layer in a pane allows for more control over individual layers because leaflet
creates separate HTML elements for each pane. They can be referenced by the class name `leaflet-{pane label}-pane`. For example, the basemap pane can be selected with `leaflet-basemap-pane`.

Closes #52 

### Demo

<img width="934" alt="Screen Shot 2022-09-02 at 11 41 59 AM" src="https://user-images.githubusercontent.com/8356789/188190187-9cfe4d83-d05d-47a9-94bf-b96459b8bc9f.png">

### Notes

I was not able to get the reference layer to use a pane. I tried wrapping it in a `featureGroup` like the user polygon layer, but removed all the edit controls and the image was non-interactive. For now, it seems to work well enough to place all the other layers around that one.

TODO:  From #52: 
> When multiple are visible, the topmost one should receive all mouse events

As is, things seem to be working well. Edit handles can be selected over other layers. The only problem is that you cannot add a polygon over the reference image. This is probably due to this line from Leaflet Distortable Image:
https://github.com/publiclab/Leaflet.DistortableImage/blob/main/dist/leaflet.distortableimage.css#L6


## Testing Instructions

Adding this feature allows the layers to be ordered based on their zLevel. Previously the municipal boundaries layer would show above the reference image. To test this:

- Toggle both the reference image and the municipal boundaries layers on
- Move the reference image to some municipal boundaries and ensure they appear above them.

## Checklist
